src/main/java/frc/team2412/robot/subsystems/DrivebaseSubsystem.java

```diff
@@ -20,145 +16,190 @@
 
-	private AHRS gyroscope;
+	private Gyroscope gyroscope;
 
 	private SwerveDriveOdometry odometry;
 	private Pose2d pose;
 
 	private Field2d field = new Field2d();
 
 	public DrivebaseSubsystem() {
-		gyroscope = new AHRS(SerialPort.Port.kMXP);
+		gyroscope = IS_COMP ? new Pigeon2Gyro(Hardware.GYRO_PORT) : new NavXGyro(SerialPort.Port.kMXP);
 
-		odometry =
-				new SwerveDriveOdometry(
-						kinematics, Rotation2d.fromDegrees(gyroscope.getYaw()), getModulePositions());
+		odometry = new SwerveDriveOdometry(kinematics, gyroscope.getAngle(), getModulePositions());
 		pose = odometry.getPoseMeters();
 
 		balanceController =
-				PFFController.ofDouble(tipF, tipP)
-						.setTargetPosition((double) gyroscope.getRoll())
-						.setTargetPositionTolerance(tipTolerance);
+				PFFController.ofDouble(TIP_F, TIP_P)
+						.setTargetPosition(gyroscope.getRawRoll().getDegrees())
+						.setTargetPositionTolerance(TIP_TOLERANCE);
 
 		// configure encoders offsets
 		for (int i = 0; i < moduleEncoders.length; i++) {
 			moduleEncoders[i].configFactoryDefault();
+			moduleEncoders[i].configSensorInitializationStrategy(
+					SensorInitializationStrategy.BootToAbsolutePosition);
 		}
 
 		// configure drive motors
 		for (int i = 0; i < moduleDriveMotors.length; i++) {
-			WPI_TalonFX driveMotor = moduleDriveMotors[i];
-			driveMotor.setNeutralMode(NeutralMode.Brake);
-			driveMotor.setSensorPhase(true);
-
-			driveMotor.configNeutralDeadband(0.01);
-			driveMotor.configSelectedFeedbackSensor(
-					TalonFXFeedbackDevice.IntegratedSensor, talonFXLoopNumber, canTimeoutMS);
-
-			driveMotor.config_kP(talonFXLoopNumber, 0.1);
-			driveMotor.config_kI(talonFXLoopNumber, 0.001);
-			driveMotor.config_kD(talonFXLoopNumber, 1023.0 / 20660.0);
+			MotorController driveMotor = moduleDriveMotors[i];
+			driveMotor.setNeutralMode(MotorController.MotorNeutralMode.BRAKE);
+
+			if (IS_COMP) {
+				driveMotor.setControlMode(MotorControlMode.VOLTAGE);
+			} else {
+				driveMotor.setControlMode(MotorControlMode.VELOCITY);
+				driveMotor.setPID(0.1, 0.001, 1023.0 / 20660.0);
+			}
 		}
 
 		// configure angle motors
 		for (int i = 0; i < moduleAngleMotors.length; i++) {
-			WPI_TalonFX steeringMotor = moduleAngleMotors[i];
+			MotorController steeringMotor = moduleAngleMotors[i];
 			steeringMotor.configFactoryDefault();
-			steeringMotor.configSelectedFeedbackSensor(
-					TalonFXFeedbackDevice.IntegratedSensor, talonFXLoopNumber, canTimeoutMS);
+			steeringMotor.setNeutralMode(MotorNeutralMode.COAST);
 			// Configure PID values
-			steeringMotor.config_kP(talonFXLoopNumber, 0.15, canTimeoutMS);
-			steeringMotor.config_kI(talonFXLoopNumber, 0.00, canTimeoutMS);
-			steeringMotor.config_kD(talonFXLoopNumber, 1.0, canTimeoutMS);
-			steeringMotor.setSelectedSensorPosition(
-					getModuleAngles()[i].times(((ticksPerRotation / 360) * steerReduction)).getDegrees());
+			if (IS_COMP) {
+				steeringMotor.setPID(0.15, 0, 0);
```

Comment:
Please add a todo or issue reference to tune this. The needs to be a D value to minimize overshooting (the I value of 0 is correct, as this is controlling position).

Reply:
added in #33